SYCL/ESIMD/api/functional/ctors/ctor_broadcast_fp_extra.cpp

```diff
@@ -0,0 +1,60 @@
+//==------- ctor_broadcast_fp_extra.cpp  - DPC++ ESIMD on-device test ------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu, level_zero
+// XREQUIRES: gpu
+// TODO gpu and level_zero in REQUIRES due to only this platforms supported yet.
+// The current "REQUIRES" should be replaced with "gpu" only as mentioned in
+// "XREQUIRES".
+// UNSUPPORTED: cuda, hip
+// RUN: %clangxx -fsycl %s -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// XFAIL: *
+// TODO The simd filled with unexpected values if the source type is floating
+// point type.
+//
+// Test for simd broadcast constructor.
+// This test uses fp extra data types, dimensionality and different simd
+// constructor invocation contexts.
+// Type of a value that will be provided to the broadcast constructor may be
+// differ, than value, that will be provided to the simd when it will be
+// constructed. It is expected for a new simd instance to store same data as the
+// one passed as the source simd constructor.
+
+#include "ctor_broadcast.hpp"
+
+using namespace sycl::ext::intel::experimental::esimd;
+using namespace esimd_test::api::functional;
+
+int main(int, char **) {
+  sycl::queue queue(esimd_test::ESIMDSelector{},
+                    esimd_test::createExceptionHandler());
+
+  bool passed = true;
+
+  const auto uint_types = get_tested_types<tested_types::uint>();
+  const auto sint_types = get_tested_types<tested_types::sint>();
+  const auto fp_extra_types = get_tested_types<tested_types::fp_extra>();
+  using use_ref_conv_values = std::false_type;
+  const auto single_dim = get_dimensions<8>();
+  const auto vardecl_context = unnamed_type_pack<ctors::var_decl>::generate();
```

Comment:
```suggestion
  const auto context = unnamed_type_pack<ctors::var_decl>::generate();
```
Nit: I would suggest to make names a bit more generic. Otherwise how many renames will you do if you select a different value to store?

Reply:
this comment applied [here](https://github.com/intel/llvm-test-suite/pull/690/commits/da1b71a5bd6811a44a5ea48d03bb911518fe1655#diff-e3103512a9fb456b0e983db0746b4c64e4ae20d76e626d6ae2eb93293aa0056dR44)